pkgs/by-name/ne/nexusmods-app/package.nix

```diff
@@ -1,114 +1,167 @@
-       ])))
-    ])
-  ];
+  #FIXME!!
+  # doCheck = true;
+  doCheck = false;
+
+  dotnetTestFlags =
+    let
+      filters =
+        [
+          "Category!=Disabled"
+          "FlakeyTest!=True"
+          "RequiresNetworking!=True"
+        ]
+        ++ lib.map (n: "FullyQualifiedName!=${n}") (
+          [
+            "NexusMods.UI.Tests.ImageCacheTests.Test_LoadAndCache_RemoteImage"
+            "NexusMods.UI.Tests.ImageCacheTests.Test_LoadAndCache_ImageStoredFile"
+
+            # TODO: investigate these test fails...
+            # Looks like the tests are still trying to use the bundled 7zz?
+            #
+            # Failed to start a process with file path '/build/source/tests/NexusMods.DataModel.Tests/bin/Release/net8.0/runtimes/linux-x64/native/7zz'.
+            # Target file or working directory doesn't exist, or the provided credentials are invalid.
+
+            # Failed with No Extractors found for file data_7zip_lzma2.7z
+            "NexusMods.CLI.Tests.VerbTests.ExtractArchive.CanExtractArchive"
+            "NexusMods.CLI.Tests.VerbTests.AnalyzeArchive.CanAnalyzeArchives"
+
+            # Failed with No Extractors found for file StubbedGameState.zip
+            "NexusMods.CLI.Tests.VerbTests.ModManagementVerbs.CanCreateAndManageLists"
+            "NexusMods.DataModel.Tests.ToolTests.CanRunTools"
+            "NexusMods.DataModel.Tests.FileHashCacheTests.UpdatingHashesReturnsTheUpdate"
+            "NexusMods.DataModel.Tests.FileHashCacheTests.CanGetHashOfSingleFile"
+            "NexusMods.DataModel.Tests.FileHashCacheTests.CanHashFolder"
+            "NexusMods.DataModel.Tests.FileOriginRegistryTests.RegisterFolder_WhenCalledTwice_ShouldOnlyAppendNewData"
+          ]
+          ++ lib.optionals (!enableUnfree) [
+            "NexusMods.Games.FOMOD.Tests.FomodXmlInstallerTests.InstallsFilesSimple_UsingRar"
+          ]
+        );
+    in
+    [
+      "--environment=USER=nobody"
+      "--filter=${lib.strings.concatStringsSep "&" filters}"
+    ];
 
   passthru = {
     tests = {
       serve = runCommand "${pname}-test-serve" { } ''
-        ${nexusmods-app}/bin/${nexusmods-app.meta.mainProgram}
+        ${nexusmods-app}/bin/${meta.mainProgram}
         touch $out
       '';
       help = runCommand "${pname}-test-help" { } ''
-        ${nexusmods-app}/bin/${nexusmods-app.meta.mainProgram} --help
+        ${nexusmods-app}/bin/${meta.mainProgram} --help
         touch $out
       '';
       associate-nxm = runCommand "${pname}-test-associate-nxm" { } ''
-        ${nexusmods-app}/bin/${nexusmods-app.meta.mainProgram} associate-nxm
+        ${nexusmods-app}/bin/${meta.mainProgram} associate-nxm
         touch $out
       '';
       list-tools = runCommand "${pname}-test-list-tools" { } ''
-        ${nexusmods-app}/bin/${nexusmods-app.meta.mainProgram} list-tools
+        ${nexusmods-app}/bin/${meta.mainProgram} list-tools
```

Comment:
metaMainprogram should not really be used like this. It falls under the same category as pname. Just write the bvinary name out.

Reply:
I don't mind doing this, but could you clarify the reason why this is bad practice?

From my perspective using `meta.mainProgram` here is DRY-er. Personally I'd go one step further and use `lib.getExe`, which does the same thing under the hood.